sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezDataAggregator.java

```diff
@@ -0,0 +1,212 @@
+import io.opentelemetry.trace.Status.CanonicalCode;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A data aggregator for the traceZ zPage.
+ *
+ * <p>The traceZ data aggregator compiles information about the running spans, span latencies, and
+ * error spans for the frontend of the zPage.
+ */
+@ThreadSafe
+final class TracezDataAggregator {
+  private final TracezSpanProcessor spanProcessor;
+
+  /**
+   * Constructor for {@link TracezDataAggregator}.
+   *
+   * @param spanProcessor collects span data.
+   */
+  public TracezDataAggregator(TracezSpanProcessor spanProcessor) {
+    this.spanProcessor = spanProcessor;
+  }
+
+  /**
+   * Returns a Set of running and completed span names for {@link TracezDataAggregator}.
+   *
+   * @return a Set of {@link String}.
+   */
+  public Set<String> getSpanNames() {
+    Set<String> spanNames = new TreeSet<>();
+    Collection<ReadableSpan> allRunningSpans = spanProcessor.getRunningSpans();
+    for (ReadableSpan span : allRunningSpans) {
+      spanNames.add(span.getName());
+    }
+    spanNames.addAll(spanProcessor.getCompletedSpanCache().keySet());
+    return spanNames;
+  }
+
+  /**
+   * Returns a Map of the running span counts for {@link TracezDataAggregator}.
+   *
+   * @return a Map of span counts for each span name.
+   */
+  public Map<String, Integer> getRunningSpanCounts() {
+    Collection<ReadableSpan> allRunningSpans = spanProcessor.getRunningSpans();
+    Map<String, Integer> numSpansPerName = new HashMap<>();
+    for (ReadableSpan span : allRunningSpans) {
+      Integer prevValue = numSpansPerName.get(span.getName());
+      numSpansPerName.put(span.getName(), prevValue != null ? prevValue + 1 : 1);
+    }
+    return numSpansPerName;
+  }
+
+  /**
+   * Returns a List of all running spans with a given span name for {@link TracezDataAggregator}.
+   *
+   * @param spanName name to filter returned spans.
+   * @return a List of {@link SpanData}.
+   */
+  public List<SpanData> getRunningSpans(String spanName) {
+    Collection<ReadableSpan> allRunningSpans = spanProcessor.getRunningSpans();
+    List<SpanData> filteredSpans = new ArrayList<>();
+    for (ReadableSpan span : allRunningSpans) {
+      if (span.getName().equals(spanName)) {
+        filteredSpans.add(span.toSpanData());
+      }
+    }
+    return filteredSpans;
+  }
+
+  /**
+   * Returns a Map of counts for the {@link Status#OK} spans within [lowerBound, upperBound) {@link
+   * TracezDataAggregator}.
+   *
+   * @param lowerBound latency lower bound (inclusive)
+   * @param upperBound latency upper bound (exclusive)
+   * @return a Map of span counts for each span name within the bounds.
+   */
+  public Map<String, Integer> getSpanLatencyCounts(long lowerBound, long upperBound) {
```

Comment:
This method isn't being used. Can we remove it?

Reply:
Removed it